codex-rs/core/src/exec_policy.rs

```diff
@@ -129,9 +127,9 @@ pub(crate) fn default_policy_path(codex_home: &Path) -> PathBuf {
 
 pub(crate) async fn append_execpolicy_amendment_and_update(
```

Comment:
This change concerns me: it derives a new `ExecPolicy`, but there could be clients with a reference to the original one and now it would be out of date, no?